tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4217,6 +4217,8 @@ void Parser::ParseDeclarationSpecifiers(DeclSpec &DS,
         isInvalid = true;
         break;
       };
+      ParseOpenCLQualifiers(DS.getAttributes()); // HLSL Change -
```

Comment:
I don't think you need to bring this line up. It shouldn't have been possible to hit.

Reply:
The conundrum is that clang recognizes that it is impossible, but GCC does not. I'd like to enable warnings as errors on GCC too. So this is toward that end.

Reply:
Bringing the break up to fix the warning seems correct, but the `ParseOpenCLQualifiers` shouldn't need to be copied up.